services/blockstorage/test/block_sync_server_test.go

```diff
@@ -0,0 +1,129 @@
+		_, err := harness.blockStorage.HandleBlockAvailabilityRequest(msg)
+
+		require.NoError(t, err, "expecting a happy flow")
+		harness.verifyMocks(t, 1)
+	})
+}
+
+func TestSourceIgnoresSendBlockAvailabilityRequestsIfFailedToRespond(t *testing.T) {
+
+	test.WithContext(func(ctx context.Context) {
+		harness := newHarness(ctx)
+		harness.setupSomeBlocks(3)
+
+		harness.gossip.When("SendBlockAvailabilityResponse", mock.Any).Return(nil, errors.New("gossip failure")).Times(1)
+		msg := builders.BlockAvailabilityRequestInput().
+			WithFirstBlockHeight(1).
+			WithLastCommittedBlockHeight(primitives.BlockHeight(2)).
+			WithLastBlockHeight(primitives.BlockHeight(2)).
+			Build()
+		_, err := harness.blockStorage.HandleBlockAvailabilityRequest(msg)
+
+		require.Error(t, err, "expecting an error from the server event flow")
+
+		harness.verifyMocks(t, 1)
+	})
+}
+
+func TestSourceRespondsWithChunks(t *testing.T) {
+	test.WithContext(func(ctx context.Context) {
+		batchSize := uint32(10)
+		harness := newHarness(ctx).withBatchSize(batchSize).withNodeKey(keys.Ed25519KeyPairForTests(4).PublicKey())
+		lastBlock := 12
+		harness.setupSomeBlocks(lastBlock)
+
+		firstHeight := primitives.BlockHeight(1)
+		lastHeight := primitives.BlockHeight(10) // hardcoding this, but it is a function of the batchSize
+
+		msg := builders.BlockSyncRequestInput().
+			WithSenderPublicKey(keys.Ed25519KeyPairForTests(1).PublicKey()).
+			WithFirstBlockHeight(firstHeight).
+			Build()
+
+		chunksResponseVerifier := func(i interface{}) bool {
+			response, ok := i.(*gossiptopics.BlockSyncResponseInput)
+			if !ok {
+				require.Failf(t, "response type does not match", "", i)
+			}
+			require.EqualValues(t, batchSize, len(response.Message.BlockPairs), "batch size does not match config")
+			require.Equal(t, keys.Ed25519KeyPairForTests(1).PublicKey(), response.RecipientPublicKey, "recipient pk is incorrect")
+			require.Equal(t, firstHeight, response.Message.SignedChunkRange.FirstBlockHeight(), "first block height mismatch")
+			require.Equal(t, lastHeight, response.Message.SignedChunkRange.LastBlockHeight(), "last block height mismatch")
+			require.Equal(t, primitives.BlockHeight(lastBlock), response.Message.SignedChunkRange.LastCommittedBlockHeight(), "last committed block height mismatch")
+			require.Equal(t, keys.Ed25519KeyPairForTests(4).PublicKey(), response.Message.Sender.SenderPublicKey(), "sender does not match config")
+			require.Equal(t, msg.Message.SignedChunkRange.BlockType(), response.Message.SignedChunkRange.BlockType(), "block type does not match the request")
+
+			return true
+		}
+
+		harness.gossip.When("SendBlockSyncResponse", mock.AnyIf("reponse should hold correct blocks", chunksResponseVerifier)).Return(nil, nil).Times(1)
+		harness.blockStorage.HandleBlockSyncRequest(msg)
+		harness.verifyMocks(t, 1)
+	})
+}
+
+func TestSourceIgnoresBlockSyncRequestIfSourceIsBehindOrInSync(t *testing.T) {
```

Comment:
`OrInSync`? what does that mean?

Reply:
It means that the source has block height less or equal to block height of the petitioner.

Reply:
so because source is behind it *may* be in sync, might be TMI because we do not really have a way to check if the source is syncing right now, and we do not care either, right? we do not 'deny syncing' as a potential source because we are syncing right now on newer blocks

Reply:
so can you suggest a better name for this test?

Reply:
yeah, just without the OrInSync
already renamed, didn't push yet